pkg/efscreate/efs.go

```diff
@@ -0,0 +1,354 @@
+	}
+	return err
+}
+
+func (efs *EFS) deleteMountTarget() error {
+	for _, mt := range efs.resources.mountTargets {
+		log("Deleting mount target id: %s\n", mt)
+		deleteTargetInput := &awsefs.DeleteMountTargetInput{
+			MountTargetId: aws.String(mt),
+		}
+		_, err := efs.efsClient.DeleteMountTarget(deleteTargetInput)
+		if err != nil {
+			return fmt.Errorf("failed to delete mount target: %v", err)
+		}
+		log("successfully deleted mount target %s", mt)
+	}
+	return nil
+}
+
+func log(msg string, args ...interface{}) {
+	msgString := fmt.Sprintf("%s\n", msg)
+	fmt.Printf(msgString, args...)
+}
+
+func (efs *EFS) deleteEFS() error {
+	backoff := wait.Backoff{
+		Duration: volumeCreateInitialDelay,
+		Factor:   operationBackoffFactor,
+		Steps:    operationRetryCount,
+	}
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		deleteEFSInput := &awsefs.DeleteFileSystemInput{FileSystemId: aws.String(efs.resources.efsID)}
+		_, delError := efs.efsClient.DeleteFileSystem(deleteEFSInput)
+		if delError != nil {
+			log("error deleting filesystem %s: %v", efs.resources.efsID, delError)
+			return false, nil
+		}
+		log("successfully deleted filesystem %s", efs.resources.efsID)
+		return true, nil
+	})
+	return err
+}
+
+func (efs *EFS) deleteSecurityGroup() error {
+	backoff := wait.Backoff{
+		Duration: operationDelay,
+		Factor:   operationBackoffFactor,
+		Steps:    operationRetryCount,
+	}
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		deleteGroupInput := &ec2.DeleteSecurityGroupInput{GroupId: aws.String(efs.resources.securityGrouID)}
+		_, delError := efs.client.DeleteSecurityGroup(deleteGroupInput)
+		if delError != nil {
+			log("error deleting security group %s: %v", efs.resources.securityGrouID, delError)
+			return false, nil
+		}
+		log("successfully deleted securityGroup %s", efs.resources.securityGrouID)
+		return true, nil
+	})
+	return err
+
+}
+
+func (efs *EFS) createEFSFileSystem() (string, error) {
+	volumeName := fmt.Sprintf(efsVolumeNameFormat, efs.infra.Status.InfrastructureName)
+	input := &awsefs.CreateFileSystemInput{
+		Encrypted:       aws.Bool(true),
+		PerformanceMode: aws.String(awsefs.PerformanceModeGeneralPurpose),
+		Tags: []*awsefs.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(volumeName),
+			},
+			{
+				Key:   aws.String(efs.getClusterTagKey()),
+				Value: aws.String("owned"),
+			},
+		},
+	}
+	response, err := efs.efsClient.CreateFileSystem(input)
+	if err != nil {
+		log("error creating filesystem: %v", err)
+		return "", fmt.Errorf("error creating filesystem: %v", err)
+	}
+	err = efs.waitForEFSToBeAvailable(*response.FileSystemId)
+	if err != nil {
+		log("error waiting for filesystem to become available: %v", err)
+		return *response.FileSystemId, err
+	}
+	return *response.FileSystemId, nil
+}
+
+func (efs *EFS) createMountTargets(efsID string, sgID string) ([]string, error) {
+	var mountTargets []string
+	for i := range efs.subnetIDs {
```

Comment:
Nit: it looks odd that some EFS objects are passed as parameters and other is passed via `efs`. There already is `efs.resources.efsID` and `securityGroupID`. This applies for most of the code, here it's the most visible.